### PR TITLE
move base.table.max-transaction-id from table property to partitionProperties

### PR DESCRIPTION
**Pull request by hzluting**
## Why are the changes needed?
  1. make partition properties to unified saved.

## Brief change log
  - *move base.table.max-transaction-id from table property to partitionProperties*

### CODE REVIEW

File: hive/src/main/java/com/netease/arctic/hive/op/HiveOperationTransaction.java [hunk 1]
```diff
@@ -173,182 +158,4 @@ public void commit() {
       }
     }
   }
-
-  private class TransactionalTable implements Table {
```
Comment (zhoujinsong) [inline hive/src/main/java/com/netease/arctic/hive/op/HiveOperationTransaction.java:177]: Why remove this `TransactionalTable` implementation? I think we need overwride transactional table `newOverwrite` and `newReplacePartitions` method?

**Review by zhoujinsong — CHANGES_REQUESTED** [review]
@hzluting I left some comment. PTAL.

File: hive/src/main/java/com/netease/arctic/hive/op/HiveOperationTransaction.java (hunk 1 above)
Comment (hzluting) [reply hive/src/main/java/com/netease/arctic/hive/op/HiveOperationTransaction.java:177]: First, I find no use the TransactionalTable anywhere before.
I think use the transactionalTable in HiveOperationTransaction.wrapped is enough?

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**